main.py

```diff
@@ -854,6 +856,27 @@ def run(
                 break
 
 
+def run_async_update_with_retries(mgr, current_block, logging_path, max_retries=5):
+    failed_async_calls = 0
+
+    while failed_async_calls < max_retries:
+        try:
+            async_update_pools_from_contracts(mgr, current_block, logging_path)
+            return  # Successful execution
+        except AyncUpdateRetryException as e:
+            failed_async_calls += 1
+            mgr.update_remaining_pools()
+            mgr.cfg.logger.error(f"Attempt {failed_async_calls} failed: {e}")
```

Comment:
Perhaps move this logging message to be printed prior to calling the `update_remaining_pools` function, in case this function itself throws an exception.

Reply:
fixed